base/functors.jl

```diff
@@ -61,6 +61,20 @@ end
 call(f::UnspecializedFun{1}, x) = f.f(x)
 call(f::UnspecializedFun{2}, x, y) = f.f(x,y)
 
+# Special purpose functors
+
+type Predicate <: Func{1}
+    f::Function
+end
```

Comment:
With such definition, we are not able to encapsulate a callable *functor* (not *function*) into an instance of `Predicate`.

For example:
```julia
immutable IsPosFun end
call(::IsPosFun, x) = (x > zero(x))

# The following statement would cause an error
Predicate(IsPosFun())
```

Use of functors is quite important, especially in cases where performance is critical.